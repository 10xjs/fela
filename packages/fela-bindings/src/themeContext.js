/* @flow */
import ThemeFactory from './ThemeFactory'
import ThemeProviderFactory from './ThemeProviderFactory'

const defaultTheme = {}

export default function themeFactory(
  createElement: Function,
  createContext: Function
) {
  const { Provider, Consumer } = createContext(defaultTheme)

  return {
    ThemeProvider: ThemeProviderFactory(createElement, Consumer, Provider),
    Theme: ThemeFactory(createElement, Consumer),
  }
}
