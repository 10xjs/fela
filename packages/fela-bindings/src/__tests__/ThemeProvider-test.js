import React, { createElement, createContext } from 'react'
import TestRenderer from 'react-test-renderer'

import themeContext from '../themeContext'

const { ThemeProvider, Theme } = themeContext(createElement, createContext)

describe('Using the ThemeProvider', () => {
  it('should pass the theme via context', () => {
    const wrapper = TestRenderer.create(
      <ThemeProvider theme={{ color: 'red' }}>
        <Theme>{theme => <div>{JSON.stringify(theme)}</div>}</Theme>
      </ThemeProvider>
    )

    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  it('should update child nodes if the theme updates', () => {
    const wrapper = TestRenderer.create(
      <ThemeProvider theme={{ color: 'red' }}>
        <Theme>{theme => <div>{JSON.stringify(theme)}</div>}</Theme>
      </ThemeProvider>
    )

    // TODO: use enzyme as soon as React 16.3 context API is merged
    // wrapper.setProps({ theme: { color: 'blue' } })

    expect(wrapper.toJSON).toMatchSnapshot()
  })
})
