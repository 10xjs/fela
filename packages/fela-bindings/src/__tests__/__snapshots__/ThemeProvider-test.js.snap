// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Using the ThemeProvider should pass the theme via context 1`] = `
<div>
  {"color":"red"}
</div>
`;

exports[`Using the ThemeProvider should update child nodes if the theme updates 1`] = `[Function]`;
